### PR TITLE
Update Release workflow and update examples

### PR DESCRIPTION
**Pull request by sahanHe**
# Description

$subject
related to : [https://github.com/ballerina-platform/ballerina-library/issues/6080](https://github.com/ballerina-platform/ballerina-library/issues/6080)

This PR will migrate PR build and publish release workflow to centralised one, as an additional task to enable buildable samples in the repo, a new set of examples have been added

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,62 +1,20 @@
...
+    if: ${{ github.repository_owner == 'ballerina-platform' }}
+    uses: ballerina-platform/ballerina-standard-library/.github/workflows/release-package-connector-template.yml@main
+    secrets: inherit
+    with:
+      package-name: salesforce
+      package-org: ballerinax
+      additional-build-flags: "-x :salesforce-examples:build"
+      additional-release-flags: "-x :salesforce-examples:build"
+      additional-publish-flags: "-x :salesforce-examples:build"
+
```
Comment (MohamedSabthar) [inline .github/workflows/release.yml:20]:
```suggestion

```

**Review by MohamedSabthar — APPROVED** [review]

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,62 +1,20 @@
-name: Deployment
+name: Publish Release
 
 on:
-  release:
-    types: [ published ]
+  workflow_dispatch:
+  repository_dispatch:
+    types: [ stdlib-release-pipeline ]
```
Comment (NipunaRanasinghe) [inline .github/workflows/release.yml:6]: is it correct to have `stdlib-release-pipeline` repository dispatch as a release trigger for the connector modules? @MohamedSabthar @ThisaruGuruge any thoughts?

File: settings.gradle [hunk 3]
```diff
@@ -30,7 +30,7 @@ include ":${projectName}-examples"
 project(':checkstyle').projectDir = file("build-config${File.separator}checkstyle")
 project(":${projectName}-native").projectDir = file('native')
 project(":${projectName}-ballerina").projectDir = file('ballerina')
-project(":${projectName}-examples").projectDir = file('examples')
+project(":${projectName}-examples").projectDir = file('samples')
```
Comment (NipunaRanasinghe) [inline settings.gradle:33]: our connector convention is to use a single directory called `examples` to house all the samples related to the particular module. Is there any reason to deviate from that?

Comment (sahanHe) [reply settings.gradle]: This was corrected in [db871a1](https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/322/commits/db871a17080bb612b6313fc1db13a6ae72c0d39f).

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -1,62 +1,19 @@
...
+    name: Run Release Workflow
+    if: ${{ github.repository_owner == 'ballerina-platform' }}
+    uses: ballerina-platform/ballerina-standard-library/.github/workflows/release-package-connector-template.yml@main
+    secrets: inherit
+    with:
+      package-name: salesforce
+      package-org: ballerinax
+      additional-build-flags: "-x :salesforce-examples:build"
+      additional-release-flags: "-x :salesforce-examples:build"
+      additional-publish-flags: "-x :salesforce-examples:build"
```
Comment (sahanHe) [inline .github/workflows/release.yml:19]:
```suggestion
      additional-publish-flags: "-x :salesforce-examples:build"
      
```

File: .github/workflows/build-with-bal-test-graalvm.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+jobs:
+  call_stdlib_workflow:
+    name: Run StdLib Workflow
+    if: ${{ github.event_name != 'schedule' || (github.event_name == 'schedule' && github.repository_owner == 'ballerina-platform') }}
+    uses: ballerina-platform/ballerina-standard-library/.github/workflows/build-with-bal-test-graalvm-connector-template.yml@main
+    secrets: inherit
+    with:
+      additional-build-flags: "-x :salesforce-examples:build"
+
```
Comment (aashikam) [inline .github/workflows/build-with-bal-test-graalvm.yml:20]:
```suggestion
      
      
```

File: .github/workflows/daily-build.yml [hunk 6]
```diff
@@ -2,71 +2,14 @@ name: Daily build
...
-            }
-          }"
+  call_workflow:
+    name: Run Daily Build Workflow
+    if: ${{ github.repository_owner == 'ballerina-platform' }}
+    uses: ballerina-platform/ballerina-standard-library/.github/workflows/daily-build-connector-template.yml@main
+    secrets: inherit
+    with:
+      repo-name: module-ballerinax-sfdc
+
```
Comment (aashikam) [inline .github/workflows/daily-build.yml:15]:
```suggestion
      
      
```
saw few more places

**Comment by aashikam** [comment]
this `baseClient` should be `salesforce` https://github.com/ballerina-platform/module-ballerinax-sfdc/blob/b99da1f0f594eecf93e62668883c1c1cdec9d7ba/README.md?plain=1#L100

File: .github/workflows/release.yml (hunk 2 above)
Comment (MohamedSabthar) [reply .github/workflows/release.yml:6]: We can remove this.

**Comment by sahanHe** [comment]
> this `baseClient` should be `salesforce`
> 
> https://github.com/ballerina-platform/module-ballerinax-sfdc/blob/b99da1f0f594eecf93e62668883c1c1cdec9d7ba/README.md?plain=1#L100

this was updated

File: .github/workflows/release.yml (hunk 2 above)
Comment (sahanHe) [reply .github/workflows/release.yml]: removed [70dfeb8](https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/322/commits/70dfeb8885ce88fef103f10126fbef96e80e9ff2)

File: examples/apex_rest_api_usecases/create_case/Ballerina.toml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+[package]
+org = "salsforce_exsamples"
```
Comment (Nuvindu) [inline examples/apex_rest_api_usecases/create_case/Ballerina.toml:2]:
```suggestion
org = "salsforce_examples"
```

File: examples/bulk_api_usecases/execute_update_job/Ballerina.toml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+[package]
+org = "salsforce_exsamples"
```
Comment (Nuvindu) [inline examples/bulk_api_usecases/execute_update_job/Ballerina.toml]:
```suggestion
org = "salesforce_examples"
```

Comment (Nuvindu) [reply examples/bulk_api_usecases/execute_update_job/Ballerina.toml]: check for other places as well

File: examples/rest_api_usecases/delete_sobject/README.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+### 1. Set up
+Refer to the setup guide in [ReadMe](../../../README.md) for necessary credentials.
+
+### 2. Configuration
+
+Configure Salesforce API credentials in Config.toml in the example directory:
+
+```toml
+clientId = "<BASE_URL>"
```
Comment (Nuvindu) [inline examples/rest_api_usecases/delete_sobject/README.md:15]:
```suggestion
clientId = "<CLIENT_ID>"
```

Comment (Nuvindu) [reply examples/rest_api_usecases/delete_sobject/README.md:15]: check for other places

File: examples/soap_api_usecases/convert_lead/convert_lead.bal [hunk 10]
```diff
@@ -16,14 +16,13 @@
 
 import ballerinax/salesforce.soap;
 import ballerina/log;
-import ballerina/os;
 
 // Create Salesforce client configuration by reading from environemnt.
```
Comment (Nuvindu) [inline examples/soap_api_usecases/convert_lead/convert_lead.bal:20]:
```suggestion
// Create Salesforce client configuration by reading from environment.
```

Comment (Nuvindu) [reply examples/soap_api_usecases/convert_lead/convert_lead.bal:20]: check for other places

File: examples/apex_rest_api_usecases/create_case/Ballerina.toml (hunk 7 above)
Comment (sahanHe) [reply examples/apex_rest_api_usecases/create_case/Ballerina.toml]: updated all instances in [bbda6d2](https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/322/commits/bbda6d24f4c2c825f38ade4a9c7ec2467a42e73d)

File: ballerina/Package.md [hunk 11]
```diff
@@ -93,10 +93,14 @@ bal run
...
+## Examples
+
+The `salesforce` connector provides practical examples illustrating usage in various scenarios. Explore these examples below, covering use cases like creating sObjects, retrieving records, and executing bulk operations.
+
+1. [Salesforce REST API usecases](https://github.com/ballerina-platform/module-ballerinax-sfdc/tree/main/examples/rest_api_usecases) - How to employ REST API of Salesforce to carryout varies tasks.
 
-2. [Salesforce new contact to Twilio SMS](https://github.com/ballerina-platform/module-ballerinax-sfdc/tree/main/examples/salesforce-new-contact-to-twilio-sms) - This example sends a Twilio SMS for every new Salesforce contact.
+2. [Salesforce Bulk API usecases](https://github.com/ballerina-platform/module-ballerinax-sfdc/tree/main/examples/bulk_api_usecases) - How to employ Bulk API of Salesforce to execute Bulk jobs.
```
Comment (Nuvindu) [inline ballerina/Package.md:102]:
```suggestion
2. [Salesforce Bulk API use cases](https://github.com/ballerina-platform/module-ballerinax-sfdc/tree/main/examples/bulk_api_usecases) - How to employ Bulk API of Salesforce to execute Bulk jobs.
```

File: examples/rest_api_usecases/update_sobject/README.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+
+```toml
+clientId = "<BASE_URL>"
+clientSecret = "<CLIENT_SECRET>"
+refreshToken = "<REFRESH_TOKEN>"
+refreshUrl = "<REFRESH_URL>"
+baseUrl = "<BASE_URL>"
+```
+
+## Run the Example
```
Comment (Nuvindu) [inline examples/rest_api_usecases/update_sobject/README.md:22]:
```suggestion
## Run the example
```

File: examples/apex_rest_api_usecases/create_case/Ballerina.toml (hunk 7 above)
Comment (sahanHe) [reply examples/apex_rest_api_usecases/create_case/Ballerina.toml]: Updated in [bbda6d2](https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/322/commits/bbda6d24f4c2c825f38ade4a9c7ec2467a42e73d)

File: examples/rest_api_usecases/update_sobject/README.md (hunk 12 above)
Comment (sahanHe) [reply examples/rest_api_usecases/update_sobject/README.md]: Updated in [Update code according to comments](https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/322/commits/637bff31ea6fdcb76c54d48b8f7054b644e6d85a)

File: ballerina/Package.md (hunk 11 above)
Comment (sahanHe) [reply ballerina/Package.md]: Updated in [Update code according to comments](https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/322/commits/637bff31ea6fdcb76c54d48b8f7054b644e6d85a)

File: examples/soap_api_usecases/convert_lead/convert_lead.bal (hunk 10 above)
Comment (sahanHe) [reply examples/soap_api_usecases/convert_lead/convert_lead.bal]: Updated in [Update code according to comments](https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/322/commits/637bff31ea6fdcb76c54d48b8f7054b644e6d85a)

File: examples/rest_api_usecases/delete_sobject/README.md (hunk 9 above)
Comment (sahanHe) [reply examples/rest_api_usecases/delete_sobject/README.md]: Updated in [Update code according to comments](https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/322/commits/637bff31ea6fdcb76c54d48b8f7054b644e6d85a)

**Review by NipunaRanasinghe — DISMISSED** [review]
LGTM

**Review by aashikam — DISMISSED** [review]

**Review by MohamedSabthar — APPROVED** [review]

**Review by daneshk — APPROVED** [review]

**Review by Nuvindu — APPROVED** [review]

**Review by daneshk — APPROVED** [review]

**Pull request merged**